core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/configuration/script/ScriptVersionConfiguration.java

```diff
@@ -154,14 +180,40 @@ public void insert(ScriptVersion scriptVersion) {
         getMetadataRepository().executeUpdate(getInsertStatement(scriptVersion));
     }
 
-
     @Override
     public void update(ScriptVersion scriptVersion){
         LOGGER.trace(MessageFormat.format("Updating ScriptVersion {0}-{1}.", scriptVersion.getScriptId(), scriptVersion.getNumber()));
         if (!exists(scriptVersion)) {
             throw new MetadataDoesNotExistException(scriptVersion);
         }
-        getMetadataRepository().executeUpdate(updateStatement(scriptVersion));
+        getMetadataRepository().executeUpdate(getUpdateStatement(scriptVersion));
+    }
+
+    public void restoreDeletedScriptVersion(ScriptVersionKey scriptVersionKey) {
+        LOGGER.trace(MessageFormat.format("Restoring deleted ScriptVersion {0}", scriptVersionKey.getScriptKey().getScriptId()));
+        if (exists(scriptVersionKey)) {
+            throw new MetadataAlreadyExistsException(scriptVersionKey.getScriptKey());
+        } else if (!existsDeleted(scriptVersionKey)) {
+            throw new MetadataDoesNotExistException(scriptVersionKey.getScriptKey());
+        }
+        getMetadataRepository().executeUpdate(markActiveScriptVersion(scriptVersionKey));
+        LOGGER.trace(MessageFormat.format("Successfully Restored deleted ScriptVersion {0}", scriptVersionKey.getScriptKey()));
+    }
+
+    private String markInactiveScriptVersion(ScriptVersionKey scriptVersionKey) {
+        return "UPDATE " + getMetadataRepository().getTableNameByLabel("ScriptVersions") + " SET " +
+                " DELETED_AT = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getDeletedAt()) +
+                " WHERE SCRIPT_ID = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptId()) +
+                " AND SCRIPT_VRS_NB = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptVersion()) +
+                " AND DELETED_AT = 'NA' ;";
```

Comment:
Doesn't ScriptVersionKey contain deletedAt? don't e need to use this value here?